### PR TITLE
Update Workflow Actions

### PR DESCRIPTION
**Pull request by lkiesow**
This patch updates a few operations in the GitHub Actions workflows.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,12 +4,12 @@ on: [pull_request, push]
 
 jobs:
   main:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```
Comment (LukasKalbertodt) [inline .github/workflows/ci.yml:7]: I'm curious: why this change? Do you specifically want an environment that updates itself over time?

Comment (lkiesow) [reply .github/workflows/ci.yml:7]: Yes, exactly. Right now `ubuntu-latest` is `ubuntu-20.04`, but 22.04 is already available and will become `ubuntu-latest` eventually, while 20.04 will disappear. We don't have anything particularly bound to the OS and it should be fine and is unlikely to break if we auto-update.

Comment (LukasKalbertodt) [reply .github/workflows/ci.yml:7]: Oh yeah that makes sense!
